### PR TITLE
upgrade to polkadot v1.0.0. bump 1.5.1, runtime v16

### PR DESCRIPTION
**Pull request by brenzi**
this upgrade also bumps pallet deps and therefore includes the following PRs:
* https://github.com/encointer/pallets/pull/345
* https://github.com/encointer/pallets/pull/346

bump spec version of encointer-runtime to v16 and collator crate version to 1.5.1

### CODE REVIEW

File: polkadot-parachains/encointer-runtime/src/xcm_config.rs [hunk 1]
```diff
@@ -218,6 +218,8 @@ impl pallet_xcm::Config for Runtime {
 	type ReachableDest = ReachableDest;
 	type AdminOrigin = EnsureRoot<AccountId>;
 	type UniversalLocation = UniversalLocation;
+	type MaxRemoteLockConsumers = ConstU32<0>;
+	type RemoteLockConsumerIdentifier = ();
```
Comment (clangenb) [inline polkadot-parachains/encointer-runtime/src/xcm_config.rs:222]: I assume that these are aligned with statemin*?

File: polkadot-parachains/src/service.rs [hunk 2]
```diff
@@ -316,51 +293,30 @@ where
...
-				client: client.clone(),
-				pool: transaction_pool.clone(),
-				backend: backend.clone(),
-				offchain_indexing_enabled,
-				deny_unsafe,
-			};
-
-			rpc_extensions(deps)
 		})
-	};
```
Comment (clangenb) [inline polkadot-parachains/src/service.rs:363]: Have you tried running this? we made here some adjustments for our custom RPCs, it seems weird to me now that we remove this.

File: polkadot-parachains/src/service.rs [hunk 3]
```diff
@@ -687,7 +643,7 @@ where
 		polkadot_config,
 		collator_options,
 		id,
-		rpc_extension_builder,
+		|_| Ok(RpcModule::new(())),
```
Comment (clangenb) [inline polkadot-parachains/src/service.rs:646]: This seems not correct to me either if we want to support the encointer-rpcs.

**Review by clangenb — COMMENTED** [review]
Only a minor thing in the service file. Looks good to me otherwise.

File: polkadot-parachains/encointer-runtime/src/xcm_config.rs (hunk 1 above)
Comment (brenzi) [reply polkadot-parachains/encointer-runtime/src/xcm_config.rs:222]: Yes. Copy paste

File: polkadot-parachains/src/service.rs (hunk 2 above)
Comment (brenzi) [reply polkadot-parachains/src/service.rs:363]: True, good catch

File: polkadot-parachains/src/service.rs (hunk 3 above)
Comment (brenzi) [reply polkadot-parachains/src/service.rs:646]: Ok. I'll check

**Comment by brenzi** [comment]
@clangenb the above remarks are a reminder to finally do proper CI here https://github.com/encointer/encointer-parachain/issues/140
;-) guess we could just start with zombienet and bootstrap_demo_community.py

File: polkadot-parachains/src/command.rs [hunk 4]
```diff
@@ -332,15 +316,7 @@ pub fn run() -> Result<()> {
...
 					if cfg!(feature = "runtime-benchmarks") {
-						runner.sync_run(|config| {
-							if config.chain_spec.is_launch() {
-								cmd.run::<Block, LaunchParachainRuntimeExecutor>(config)
-							} else if config.chain_spec.is_encointer() {
-								cmd.run::<Block, EncointerParachainRuntimeExecutor>(config)
-							} else {
-								Err("Chain doesn't support benchmarking".into())
-							}
-						})
```
Comment (brenzi) [inline polkadot-parachains/src/command.rs:343]: hmm, this remove doesn't look sane neither

Comment (brenzi) [reply polkadot-parachains/src/command.rs:343]: no, actually I think this is no longer needed as native runtime execution has been removed, right?

Comment (brenzi) [reply polkadot-parachains/src/command.rs:343]: if it should be needed, we must implement trait HostFunctions.....

Comment (clangenb) [reply polkadot-parachains/src/command.rs:343]: I was surprised too, but then I thought if the upstream passes `()`, it must just be a simplification from upstream, and as it is only the benchmarks, it does make sense to me intuitively that you actually don't need an executor.

**Review by clangenb — APPROVED** [review]
Looks goot to me now! Yeah, I agree about the CI. :)

**Pull request merged**